class-two-factor-core.php

```diff
@@ -577,43 +580,6 @@ public static function show_two_factor_login( $user ) {
 		self::login_html( $user, $login_nonce['key'], $redirect_to );
 	}
 
-	/**
-	 * Display the Backup code 2fa screen.
-	 *
-	 * @since 0.1-dev
-	 */
-	public static function backup_2fa() {
```

Comment:
The secondary provider selection doesn't work for me with this gone. I tested with Backup Codes and Dummy available, and selecting the secondary logged me in without showing the provider's UI or validating any 2FA token.

Reply:
Ah.. this is a oddity in the way the `Dummy` handler works, and something that I'd worked around in #529 via f6f5dfd0c524df5d7fc4230dfc1012cd309ab923 but failed to sync to this branch.

Basically, if you select the Dummy Auth as the provider, the auth is checked as part of the form being displayed, and well, dummy auth always returns truthful, it doesn't check to see if it's prompted and been confirmed (which IMHO, is a fault of the provider).

The reason why setting the Dummy provider as the primary provider doesn't have this problem, is that unlike the `validate_2fa` handler, there's _another_ handler that displays the form (but doesn't trigger the validation methods).

Reply:
I've merged that into this branch with 87985df9b26936a4907a0f2e60ca33ff325025ed which I think is better than keeping the existing multiple entry points.

Reply:
Okay, I was wrong.

The reason it was split was so that the handler for that action handler didn't attempt to process the auth. It duplicated most of the code, except for the code that actually performed validation.

Removing the `backup_2fa` method meant that it was processing auth on every request, which sounds fine, except that the Dummy method would always succeeded (and so then logged in), and it would cause other handlers to then instantly throw a `invalid code` error message.

In b1f6adb1e83490db4fcf7baafb4aee6f44734ea3 I've taken the alternative approach of making the singular action handle this scenario, by not processing the request if it's not an actual form submission. That removes the need for https://github.com/WordPress/two-factor/commit/f6f5dfd0c524df5d7fc4230dfc1012cd309ab923.

I'm not against restoring `backup_2fa` if you still think it's the best option.

Reply:
Nope, this sounds good, and it works well in my tests 👍🏻